### ISSUE TITLE
nvm lazy load not working

### ISSUE THREAD

**Issue by patric-eberle** [ohmyzsh/ohmyzsh#11748]
### Describe the bug

Using `zstyle ':omz:plugins:nvm' lazy yes` in `.zshrc` throws an error when opening a new terminal (and corrupts nvm auto version detection feature):

```
load-nvmrc:2: command not found: nvm_find_nvmrc
```

### Steps to reproduce

I installed oh-my-zsh on my M1 mac with

```
sh -c "$(curl -fsSL https://raw.githubusercontent.com/ohmyzsh/ohmyzsh/master/tools/install.sh)"
```

Then I installed `nvm` by documentation with

```
curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.3/install.sh | bash
```

As documented I then update the `.zshrc` configuration on/with the following lines:

```
# Which plugins would you like to load?
# Standard plugins can be found in $ZSH/plugins/
# Custom plugins may be added to $ZSH_CUSTOM/plugins/
# Example format: plugins=(rails git textmate ruby lighthouse)
# Add wisely, as too many plugins slow down shell startup.
plugins=(git nvm)

zstyle ':omz:plugins:nvm' lazy yes
zstyle ':omz:plugins:nvm' autoload yes

source $ZSH/oh-my-zsh.sh
```

Opening a new terminal tab now throws the error `load-nvmrc:2: command not found: nvm_find_nvmrc`.

Removing the `zstyle ':omz:plugins:nvm' lazy yes` line, nvm is not lazy loaded anymore, but everything else works as expected.

```
Found '/Users/*****/Projects/****/****-frontend/.nvmrc' with version <18>
Now using node v18.16.0 (npm v9.5.1)
```

### Expected behavior

The provided auto loader works without errors.

### OS / Linux distribution

macOS 13.4

### Zsh version

5.9

### Terminal emulator

iTerm2, Warp

### If using WSL on Windows, which version of WSL

None

**Comment by patric-eberle** [comment]
Makes sense, that they can not be used together, now that you mention it. It sure would be great if it would not trow an exception though. Adding a hint to the documentation would probably already help a lot :)

**Comment by thedaviddelta** [comment]
@carlosala Please add an option to silently autoload it.
With the new messages you added in this issue's fix, it reports it can't find nvm installation in my system every time I open a shell.
This is intentional, as we may not have nvm installed in every system we use but may want to have the plugin for it to work when it is.
Thanks and regards, David.

**Comment by carlosala** [comment]
hi @thedaviddelta! Which is the purpose of having the plugin without having nvm installed? I added that log to make easier to debug

**Comment by carlosala** [comment]
@thedaviddelta I see, I'll remove that log soon. Thanks for the report

**Comment by thedaviddelta** [comment]
Thank you so much!
Indeed I think it's a good idea to have some error message in some cases, so maybe it wouldn't be a bad idea to keep them, but at least giving us the option to disable them if we know what we're doing :smile:

**Comment by carlosala** [comment]
@thedaviddelta fixed in 7cf00a0b5cf0f25c16fc7902f78dff480e03fe3d!
Thanks for the report!

_gràcies :)_

**Comment by thedaviddelta** [comment]
@carlosala Gràcies a tu :smile:

**Comment by johannesrave** [comment]
heyhey,
i understand that both options set on startup in eg. `.zshrc` don't make sense. however, the behaviour i would like is:
- lazily load nvm just as with the `lazy` option, when calling eg. `pnpm` etc
- as soon as nvm has been loaded that way, autoload `.nvmrc`s when entering a directory

most of my zsh sessions would never know nvm exists, but if i need it it'll be there without me having to remember that i have to set something. is this feasible?

**Comment by carlosala** [comment]
@johannesrave I like the idea! Could you open an issue with it? Thanks!
